src/cc_impls.rs

```diff
@@ -55,23 +64,31 @@ impl<T: PartialOrd> PartialOrd for RawCc<T, O> {
     }
 }
 
-impl<T: Ord> Ord for RawCc<T, O> {
+impl<T: Ord + ?Sized> Ord for RawCc<T, O> {
     #[inline]
     fn cmp(&self, other: &RawCc<T, O>) -> Ordering {
         (**self).cmp(&**other)
     }
 }
 
-impl<T: fmt::Debug> fmt::Debug for RawCc<T, O> {
+impl<T: fmt::Debug + ?Sized> fmt::Debug for RawCc<T, O> {
     #[inline]
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "Cc({:?})", **self)
+        f.debug_tuple("Cc")
+            .field(&self.inner().deref())
+            .finish()
     }
 }
 
-impl<T: fmt::Debug> fmt::Display for RawCc<T, O> {
+impl<T: fmt::Display + ?Sized> fmt::Display for RawCc<T, O> {
     #[inline]
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         (**self).fmt(f)
     }
 }
+
+impl<T: ?Sized> fmt::Pointer for RawCc<T, O> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Pointer::fmt(&(&**self as *const T), f)
```

Comment:
I changed `(&**self as *const T)` to `self.inner().deref()` for easier understanding.